description.md

```diff
@@ -18,8 +17,34 @@ Each topic in this file contains a small description and links to the files cove
 
 ### Assignments
 
-### Drills
+Assignments are practical activities that learners undertake individually or in teams over the span of multiple weeks.
+The primary objective of an assignment is to create a compelling scenario that allows learners to apply the concepts they have learned.
+
+#### Support
+
+Assignments are typically meant for individual work, but they often require support from trainers.
+This support aims to clarify requirements and prevent students from getting stuck on tasks.
+
+Going beyond basic assistance, forums can be implemented as public platforms where learners can openly share questions, provide answers, and engage in discussions without revealing explicit solutions.
+
+#### Difficulty
+
+Each assignment should employ a gradual increase in difficulty, fostering self-evaluation and the development of critical thinking skills.
+By building a cohesive story, learners actively engage with known solutions, explore limitations, and discover suitable strategies.
+Assignments can encourage individual research by posing questions that go beyond the covered material.
+These deep-dives should be marked as **bonus tasks**, so students feel encouraged to further explore, rather than constrained.
 
+#### Project
+
+Assignments can serve as stages of a larger project.
+This encourages a structured and progressive approach to learning by understanding how building blocks come together.
+
+However, opting for a project-based approach rather than individual assignments requires additional effort and consideration. One major challenge is the strong interdependence between each stage, making it more difficult to modify or alter one stage without affecting the others.
```

Comment:
```suggestion
However, opting for a project-based approach rather than individual assignments requires additional effort and consideration.
One major challenge is the strong interdependence between each stage, making it more difficult to modify or alter one stage without affecting the others.
```